Framework/PythonInterface/plugins/algorithms/LoadWANDSCD.py

```diff
@@ -0,0 +1,169 @@
+                          + bc[::4,2::4] + bc[1::4,2::4] + bc[2::4,2::4] + bc[3::4,2::4]
+                          + bc[::4,3::4] + bc[1::4,3::4] + bc[2::4,3::4] + bc[3::4,3::4])
+                data_array[n] = bc
+                s1_array.append(f['/entry/DASlogs/HB2C:Mot:s1.RBV/average_value'].value[0])
+                duration_array.append(float(f['/entry/duration'].value[0]))
+                run_number_array.append(float(f['/entry/run_number'].value[0]))
+                monitor_count_array.append(float(f['/entry/monitor1/total_counts'].value[0]))
+
+        progress.report('Creating MDHistoWorkspace')
+        createWS_alg = self.createChildAlgorithm("CreateMDHistoWorkspace", enableLogging=False)
+        createWS_alg.setProperty("SignalInput", data_array)
+        createWS_alg.setProperty("ErrorInput", np.sqrt(data_array))
+        createWS_alg.setProperty("Dimensionality", 3)
+        createWS_alg.setProperty("Extents", '0.5,{},0.5,{},0.5,{}'.format(y_dim+0.5, x_dim+0.5, number_of_runs+0.5))
+        createWS_alg.setProperty("NumberOfBins", '{},{},{}'.format(y_dim,x_dim,number_of_runs))
+        createWS_alg.setProperty("Names", 'y,x,scanIndex')
+        createWS_alg.setProperty("Units", 'bin,bin,number')
+        createWS_alg.execute()
+        outWS = createWS_alg.getProperty("OutputWorkspace").value
+
+        progress.report('Getting IDF')
+        # Get the instrument and some logs from the first file; assume the rest are the same
+        _tmp_ws = LoadEventNexus(runs[0], MetaDataOnly=True, EnableLogging=False)
+        # The following logs should be the same for all runs
+        RemoveLogs(_tmp_ws,
+                   KeepLogs='HB2C:Mot:detz,HB2C:Mot:detz.RBV,HB2C:Mot:s2,HB2C:Mot:s2.RBV,'
+                   'HB2C:Mot:sgl,HB2C:Mot:sgl.RBV,HB2C:Mot:sgu,HB2C:Mot:sgu.RBV,'
+                   'run_title,start_time,experiment_identifier,HB2C:CS:CrystalAlign:UBMatrix',
+                   EnableLogging=False)
+
+        try:
+            ub = np.array(re.findall(r'-?\d+\.*\d*', _tmp_ws.run().getProperty('HB2C:CS:CrystalAlign:UBMatrix').value[0]),
+                          dtype=np.float).reshape(3,3)
+            sgl = np.deg2rad(_tmp_ws.run().getProperty('HB2C:Mot:sgl.RBV').value[0]) # 'HB2C:Mot:sgl.RBV,1,0,0,-1'
+            sgu = np.deg2rad(_tmp_ws.run().getProperty('HB2C:Mot:sgu.RBV').value[0]) # 'HB2C:Mot:sgu.RBV,0,0,1,-1'
+            sgl_a = np.array([[           1,            0,           0],
+                              [           0,  np.cos(sgl), np.sin(sgl)],
+                              [           0, -np.sin(sgl), np.cos(sgl)]])
+            sgu_a = np.array([[ np.cos(sgu),  np.sin(sgu),           0],
+                              [-np.sin(sgu),  np.cos(sgu),           0],
+                              [           0,            0,           1]])
+            UB = sgl_a.dot(sgu_a).dot(ub) # Apply the Goniometer tilts to the UB matrix
```

Comment:
This information must be in the documentation, to let users know that you change the UB matrix

Reply:
Added